### PR TITLE
feat(exportExcel): Generate and save excel to file system, Download generated file with token

### PR DESCRIPTION
**Pull request by JaideepPalit**
> Generate and save excel to file system, Download generated file with token.
> * Which issue is this pull request belonging to and how is it solving it? (*#1475*)
> * Did you add or update any new dependencies that are required for your change? - No

closes: #1475 

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

**Comment by smrutis1** [comment]
(from discussion): 
*  User will get an email once project spreadsheet export(i.e in `/tmp`) is completed with the download link.
*  set this `send.project.spreadsheet.export.to.mail.enabled = true` in sw360.properties file to enable this feature .

**Comment by ag4ums** [comment]
report generation is breaking when multiple users tries to generate report

**Comment by smrutis1** [comment]
To reduce the size more you can remove many columns

| project name | version | group | project type | project tag | clearingState | name | release: version | component type | clearing state | ECC information: eccStatus | ECC information: ECCN | ECC information: assessorDepartment | ECC information: assessmentDate |

**Comment by Marco-Wagner** [comment]
Yes only the mentioned colums are most important for a minimum set of an excel file.
The rest of the columns can be removed.
In my test it has reduced the size massivly.

**Comment by Marco-Wagner** [comment]
PS: I want to mention here additionally that in my test saving the exported excel file from SW360 production instance over Microsoft Excel reduces the excel file to 10% of the file size (without column change)

**Comment by Marco-Wagner** [comment]
Perhaps some excel compression option missing? :)

File: libraries/exporters/src/main/java/org/eclipse/sw360/exporter/ExcelExporter.java [hunk 1]
```diff
@@ -67,6 +78,59 @@ public InputStream makeExcelExport(List<T> documents) throws IOException, SW360E
...
+
+    public InputStream downloadExcelSheet(String token) {
+        InputStream stream = null;
+        try {
+            File file = new File(token);
+            if (file.exists()) {
+                stream = new FileInputStream(new File(token));
+            }
+        } catch (FileNotFoundException e) {
+            // logger
```
Comment (akapti) [inline libraries/exporters/src/main/java/org/eclipse/sw360/exporter/ExcelExporter.java:128]: consider adding error log here.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 2]
```diff
@@ -78,6 +78,7 @@
 import javax.servlet.http.HttpServletResponse;
 
 import java.io.IOException;
+import java.io.InputStream;
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:81]: remove unused import.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 3]
```diff
@@ -1021,23 +1043,52 @@ private void serveRemoveProject(ResourceRequest request, ResourceResponse respon
     private void exportExcel(ResourceRequest request, ResourceResponse response) {
         final User user = UserCacheHolder.getUserFromRequest(request);
         final String projectId = request.getParameter(Project._Fields.ID.toString());
-        String filename = String.format("projects-%s.xlsx", SW360Utils.getCreatedOn());
+        String token = null;
+
         try {
+            setSessionMessage(request, "Excel export has started. We will send you an email with download link once completed");
```
Comment (akapti) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java:1049]: `Excel report generation has started` seems better. If possible add localization text.

**Review by akapti — CHANGES_REQUESTED** [review]
reviewed the code. suggested some changes.

**Comment by smrutis1** [comment]
@akapti Thanks for review. Have incorporated the changes.

**Review by akapti — APPROVED** [review]
Code looks good to me.
Approved.
Tested by: @ag4ums

**Comment by smrutis1** [comment]
Resolved merge conflicts

**Pull request merged**